python/hidet/graph/ops/quant/symmetric.py

```diff
@@ -0,0 +1,77 @@
+from typing import Union
+from hidet import ir
+from hidet.ir.expr import cast, if_then_else
+from hidet.ir.compute.primitives import TensorNode, compute
+from hidet.ir import primitives as prim
+from hidet.ir.compute import cops
+
+from hidet.graph.ops.utils import Task, Operator, Tensor, input_like
+
+
+# pylint: disable=dangerous-default-value
+class SymmetricQuantizationTask(Task):
+    def __init__(self, w: TensorNode, quant_type: ir.type.DataType, dims=[-1]):
+        dims = [i if i >= 0 else len(w.shape) + i for i in dims]
+        self._assert(all(i >= 0 or i < len(w.shape) for i in dims), "dims are out of bounds")
+
+        wm = compute(
+            name='abs', shape=w.shape, fcompute=lambda *indices: if_then_else(w[indices] >= 0, w[indices], -w[indices])
+        )
+        scale = cops.reduce_cop(wm, dims, keep_dim=False, reduce_type='max')
+        scale = compute(
+            name='scaling', shape=scale.shape, fcompute=lambda *indices: quant_type.max_value / scale[indices]
+        )
+
+        def scale_weight(*indices):
+            scale_indices = [indices[i] for i in range(len(indices)) if not i in dims]
+            return cast(prim.round(w[indices] * scale[scale_indices]), quant_type)
+
+        wq = compute(name='quantize', shape=w.shape, fcompute=scale_weight)
+        super().__init__(
+            name='symmetric_quantization_task',
+            inputs=[w],
+            outputs=[wq, scale],
+            attributes={'dims': dims, 'quant_type': quant_type},
+        )
+
+
+class SymmetricDeQuantizationTask(Task):
+    def __init__(self, wq: TensorNode, scale: TensorNode, dims=[-1]):
+        dims = [i if i >= 0 else len(wq.shape) + i for i in dims]
+        self._assert(all(i >= 0 or i < len(wq.shape) for i in dims), "dims are out of bounds")
+
+        def unscale_weight(*indices):
+            scale_indices = [indices[i] for i in range(len(indices)) if not i in dims]
+            return cast(wq[indices], scale.type.dtype) / scale[scale_indices]
+
+        w = compute(name='dequantize', shape=wq.shape, fcompute=unscale_weight)
+        super().__init__(
+            name='symmetric_dequantization_task', inputs=[wq, scale], outputs=[w], attributes={'dims': dims}
```

Comment:
To be consistent with other tasks, we usually do not write the `_task` suffix to the task name. It name will be printed when compilation, and `_task` will looks redundent.